activerecord/lib/active_record/model_schema.rb

```diff
@@ -226,6 +226,7 @@ def table_name=(value)
         if defined?(@table_name)
           return if value == @table_name
           reset_column_information if connected?
+          initialize_find_by_cache
```

Comment:
Nice and simple.

I wonder: Should `reset_column_information` clear the find-by cache too?